### PR TITLE
Next (@rescript/react)

### PR DESCRIPTION
**Pull request by jeddeloh**
This branch is available via `rescript-apollo-client@next`. The only change is updating the `reason-react` dependency to `@rescript/react`. My plan was to wait for a proper release until any documentation (migration or otherwise) is up for it.

### CODE REVIEW

**Comment by jfrolich** [comment]
Great! Thanks, I probably will try this out in prod soon :)

**Comment by jeddeloh** [comment]
Awesome. I have to wait for `bs-material-ui` myself :)

**Comment by jeddeloh** [comment]
I somehow missed the docs for this: https://rescript-lang.org/docs/react/latest/introduction

Still, I would like to actually be able to run the latest myself in production, so I'll plan to release when `bs-material-ui` does https://github.com/jsiebern/bs-material-ui/issues/120

File: EXAMPLES/src/WebpackEntry.res [hunk 1]
```diff
@@ -1 +1,4 @@
-ReactDOMRe.renderToElementWithId(<App />, "root")
+switch ReactDOM.querySelector("#root") {
```
Comment (bdunn313) [inline EXAMPLES/src/WebpackEntry.res:1]: For my own curiosity can you get into why this change is necessary here? Thanks!

Comment (jeddeloh) [reply EXAMPLES/src/WebpackEntry.res:1]: `ReactDOMRe` is no longer available in `@rescript/react`. I also switched to using the pattern of loading an app pulled directly from the documentation that uses `querySelector`. If I were building a real app, I'd probably at least log on the `None` case.

Comment (bdunn313) [reply EXAMPLES/src/WebpackEntry.res:1]: Ah thanks! I missed that little update with the change!

**Review by bdunn313 — APPROVED** [review]

**Comment by jeddeloh** [comment]
Whoops, kinda forgot about this :) I'm going to publish this as latest if no one objects. I've made `bs-platform` optional in anticipation of the new `rescript` package and `melange`.
